samples/nrf_desktop/src/hw_interface/motion_simulated.c

```diff
@@ -87,6 +88,10 @@ static void generate_motion_event(void)
 
 static bool event_handler(const struct event_header *eh)
 {
+	if (atomic_get(&state, STATE_ERROR)) {
```

Comment:
I am not sure this is the right was as it causes a lot of code duplication.

Reply:
I will rethink this problem